specs/agents/tracing-instrumentation-messaging.md

```diff
@@ -0,0 +1,103 @@
+## Messaging Systems
+
+The instrumentation of messaging systems includes:
+- Transaction creation on message reception, either as a child of the message-sending span, if the sending action is traced (meaning 
+distributed tracing support), or as root
+- Span creation for message sending action
+- Span creation on message *polling that occurs within a traced transaction*
+
+### Passive vs. active message reception
+
+Message send/publish events can only be captured as spans if occurring within a traced transaction.
+Message consumption can be divided into two types: passive, where you would implement a listener that is called once a message is available, 
+and active, where the queue/topic is being polled (blocking or non-blocking). 
+
+Passive consumption typically results in a `messaging` transaction and is pretty straightforward to trace - start at entry and end at exit. 
+
+Message polling can be done within a traced transaction, in which case it should result in a messaging span, or it can be the initiating 
+event for a message handling flow. Capturing polling spans is also mostly straightforward. 
+For polling-based transactions, our goal is to capture the message handling flow, which typically *starts after the polling action exits*, 
+returning a message. This may be tricky if the handling flow is not implemented within a well defined API. Use other agent implementation
+as reference.
+The agent should not create a transaction based on polling APIs if the polling action did not result with a message (as opposed to 
+polling spans, where we want to capture such as well).
+
+### Typing
+
+- Transactions: 
+  - `transaction.type`: `messaging`
```

Comment:
We could also have `transaction.framework` here in symmetry of `span.subtype` but without being lower-cased.

Reply:
++

The intake API field for that is `transaction.context.service.framework.name` and `transaction.context.service.framework.version`